tests/components/decora_wifi/test_config_flow.py

```diff
@@ -0,0 +1,272 @@
+"""Test the decora_wifi Config Flow."""
+
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.decora_wifi.common import (
+    CommFailed,
+    DecoraWifiPlatform,
+    LoginFailed,
+)
+from homeassistant.components.decora_wifi.const import CONF_TITLE, DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+from tests.components.decora_wifi.common import MockDecoraWifiPlatform
+
+# Test inputs
+USERNAME = "username@home-assisant.com"
+UPDATED_USERNAME = "updated_username@home-assitant.com"
+PASSWORD = "test-password"
+UPDATED_PASSWORD = "updated-password"
+INCORRECT_PASSWORD = "incoreect-password"
+SCAN_INTERVAL = 120
+UPDATED_SCAN_INTERVAL = 60
+
+
+async def test_import_flow(hass: HomeAssistant):
+    """Check import flow."""
+
+    with patch.object(
+        DecoraWifiPlatform,
+        "async_setup_decora_wifi",
+        MockDecoraWifiPlatform.async_setup_decora_wifi,
+    ), patch.dict(hass.data), patch(
+        "homeassistant.components.decora_wifi.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.decora_wifi.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        hass.data[DOMAIN] = {}
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == f"{CONF_TITLE} - {USERNAME}"
+    assert result["data"] == {
+        CONF_USERNAME: USERNAME,
+        CONF_PASSWORD: PASSWORD,
+    }
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_flow(hass: HomeAssistant):
+    """Test the user flow."""
+    # Test that the flow with no user input produces a form.
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert not result["errors"]
+
+    # Test that the flow with user input calls setup entry with the appropriate data.
+    with patch.object(
```

Comment:
https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Don't patch core code. Patch the `pypi` package code instead.

Reply:
I guess in this case I assumed `test_config_flow` should only be testing the integration's `config_flow.py` module, and code imported from `common.py` should be tested separately in `test_common`.  Are the tests intended to be more integration tests rather than unit tests?

Reply:
Yes. If we test the internals it tends to break when we make changes to the entity model. We don't want 100s of integration tests that have to be updated whenever there is an internal change that does not affect how the entity is presented to the user